src/osd/OSDMap.cc

```diff
@@ -2488,6 +2498,8 @@ int OSDMap::apply_incremental(const Incremental &inc)
     stretch_mode_bucket = inc.new_stretch_mode_bucket;
   }
 
+  allow_crimson ^= inc.toggle_allow_crimson;
```

Comment:
xor should be fine assuming we really want the _toggle_ semantic and the event's life-time is one epoch only.

Reply:
The "event" is the toggle.  We will eventually want to be able to turn allow_crimson off.

Reply:
OK, I see – low-level for the future.

Reply:
I suppose I could make it an enum.

Reply:
Ok, changed it to an enum.  Preserved the ability for the Incremental application to remove the flag in case we want to backport that kind of capability later on.

Reply:
The field could be named "crimson_enabled", and just set the osdmap flag equal to it? That does mean you can't identify if a given incremental causes a change without the preceding osdmap, but is that actually a problem?

Reply:
^ I guess because that makes it hard to conditionally encode the incremental with the different struct_v?

Anyway, the enum version seems fine.